modules/git/repo_compare.go

```diff
@@ -259,3 +263,31 @@ func (repo *Repository) GetDiffFromMergeBase(base, head string, w io.Writer) err
 	}
 	return err
 }
+
+// ReadPatch will check if a diff patch exists and return stats
+func (repo *Repository) ReadPatch(prID int64) (commits string, patch []int) {
+	patchFile := fmt.Sprintf("pulls/%d.patch", prID)
+
+	stdout, err := NewCommand("apply", "--stat", patchFile).RunInDir(repo.Path)
+	if err != nil {
+		logger.Warn("No patch, or could not read patch for %v", patchFile)
+		return "", nil
+	}
+	numFile, totalAddition, totalDeletion, err := parseDiffStat(stdout)
+	if err != nil {
+		logger.Error("Could not parse Diff stat: %d, %d, %d, %v", numFile, totalAddition, totalDeletion, err)
```

Comment:
We could use this part to avoid the call in https://github.com/go-gitea/gitea/pull/17191/files#diff-9107a2d87c7913bc4fbdf735c0391d8e9ed5936ebac72a9bd23188616accc4a7R335 
But otherwise I revert to a simple read patch file